index.js

```diff
@@ -28,15 +28,21 @@ exports.register = function ( server, options, next ) {
         request.i18n.setLocale(languageCode);
       }
     }
+
+    if (pluginOptions.queryParameter && request.query && request.query[pluginOptions.queryParameter]) {
```

Comment:
We should put all the "if" statements into "else if" statements to avoid calling the setLocale multiple times. Could you do that?